Automattic/vip-governance-plugin
# Allow customizing file path for multisites and non vip sites

## Description

This will allow customizing the file path that's used by the plugin, for the rules file. This will be useful for multisites that want to serve a different file based on the site, as well as for non vip sites that don't have a private directory.

Currently, it includes just a site_id in the filter_options but there's plans for adding in more options down the line. This is built for extendability down the line for the filter.

This will also remove the need for Lodash, by porting the get method. That way we don't need a library workaround ever.

Fixes #55, #61 

## Steps to Test

- Use `vip dev-env` to create a multisite, with a site in it.
- In your private directory, create a folder and put a rules file there. Put another one outside it.
- Use the following in a plugin and activate it:
```php
add_filter('vip_governance__governance_file_path', __NAMESPACE__ . '\\customize_rules' , 10, 2 );

function customize_rules( $governance_file_path, $filter_options ) {
    if ( isset( $filter_options['site_id'] ) && $filter_options['site_id'] === 1 ) {
        return WPCOM_VIP_PRIVATE_DIR . '/' . WPCOMVIP_GOVERNANCE_RULES_FILENAME;
    } else if ( isset( $filter_options['site_id'] ) && $filter_options['site_id'] === 2 ) {
        return WPCOM_VIP_PRIVATE_DIR . '/' . 'site2/' . WPCOMVIP_GOVERNANCE_RULES_FILENAME;
    }

    return $governance_file_path;
}
```
- Ensure based on the site, the rules loaded is different.

Note: If you log in with the same `vipgo` user into the subsite, the rules are not going to be for a role that matches anything in the rules. It shall be for default use case only, as the `vipgo` user has no roles.
